lib/Sema/CSGen.cpp

```diff
@@ -2513,7 +2513,13 @@ namespace {
 
           if (!eltTy)
             return Type();
-
+          
+          // AnyPattern in the context of a tuple binding e.g. (_,_)
+          // should be allowed to bind to a hole to allow them in the context of
+          // pattern bindings like `let (_,_) = notDeclaredVar` to be a hole.
+          if (isa<AnyPattern>(eltPattern)) {
+            CS.recordAnyTypeVarAsPotentialHole(eltTy);
```

Comment:
I don't think this is necessary, @LucianoPAlmeida. Examples you have added already work correctly on main because `getTypeForPattern` called on `AnyPattern` would produce a type variable that is marked as a hole.

Could you try without your changes but with very recent main? You should see 3 fixes like this ones:

```
  [fix: ignore invalid AST node] @ locator@0x11f028a00 [Error@<stdin>:3:16]
  [fix: specify type for pattern match] @ locator@0x11f028a98 [AnyPattern@<stdin>:3:8 -> '_' pattern decl]
  [fix: specify type for pattern match] @ locator@0x11f028af0 [AnyPattern@<stdin>:3:11 -> '_' pattern decl]
```

Reply:
I can check, but while investigating this I have look at it and it was not marked... I think I've updated recently but I'll confirm that todays after work, but at top of my head I don't remember this being marked as hole for anypattern, I'll check today after work, thanks @xedin :)

Reply:
Looks like I miss that completely, was a few days behind main, so before #60203 which not sure how I miss it ... sorry for that.

Reply:
No need to apologize! It's been an active development area lately :)

Reply:
Yeah, but after looking to #60203 I also missed some other aspects in the approach, but I got it now. 
Ah, also I'm thinking in reopen this just for test cases, the new ones are slightly different from the ones from #60203 and I still like to move the previous one from `overload.swift` to `patterns.swift` but only if you think it is worth it

Reply:
Sure, sounds good to me!